data_tamer_cpp/src/channel.cpp

```diff
@@ -160,7 +161,23 @@ void LogChannel::unregister(const RegistrationID& id)
 
 void LogChannel::addDataSink(std::shared_ptr<DataSinkBase> sink)
 {
-  _p->sinks.insert(sink);
+  std::lock_guard const lock_sinks(_p->sinks_mutex);
+
+  if(!_p->logging_started)
+  {
+    _p->sinks.insert(sink);
+  }
+  else
+  {
+    sink->addChannel(_p->channel_name, _p->schema);
+    _p->sinks.insert(sink);
+  }
```

Comment:
```suggestion
  if(_p->logging_started)
  {
    sink->addChannel(_p->channel_name, _p->schema);
  }
  _p->sinks.insert(sink);
```

Could you also add a comment here noting that the logic is different if logging has started due to the `if(!_p->logging_started)` conditional code in `takeSnapshot`? It wasn't clear to me at first why the logic had to be different here.